packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -131,16 +131,20 @@ class FlutterDevice {
 
   final Device device;
   final ResidentCompiler generator;
-  List<Uri> observatoryUris;
+  Stream<Uri> observatoryUris;
   List<VMService> vmServices;
   DevFS devFS;
   ApplicationPackage package;
   List<String> fileSystemRoots;
   String fileSystemScheme;
   StreamSubscription<String> _loggingSubscription;
+  bool _isStreamingNewObservatoryUri;
   final String viewFilter;
   final bool trackWidgetCreation;
 
+  /// Whether the stream [observatoryUris] is still open.
+  bool get isStreamingNewObservatoryUri => _isStreamingNewObservatoryUri ?? false;
```

Comment:
Can this be private or `@visibleForTesting`?

Reply:
This is used from the attach.dart to exit the loop. I have renamed it to `isWaitingForObservatory` though.